rustls/src/msgs/enums.rs

```diff
@@ -197,6 +198,22 @@ impl NamedGroup {
             _ => KeyExchangeAlgorithm::ECDHE,
         }
     }
+
+    pub fn usable_for_version(&self, version: ProtocolVersion) -> bool {
+        match version {
+            ProtocolVersion::TLSv1_3 => true,
+            _ => !matches!(
```

Comment:
Makes me wonder if this could be keyed off of `SupportedProtocolVersion` instead of `ProtocolVersion`...

Reply:
At the moment `SupportedProtocolVersion` values don't get close to the callsites of this, so this would be a pretty big refactor to arrange. Will look at that in the future.